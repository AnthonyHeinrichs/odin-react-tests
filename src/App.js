import React, { useState } from "react";

const App = () => {
  const [heading, setHeading] = useState("Vancouver Canucks are going to the playoffs");

  const clickHandler = () => {
    setHeading("Just kidding");
  };

  return (
    <>
      <button type="button" onClick={clickHandler}>
        Click Me
      </button>
      <h1>{heading}</h1>
    </>
  );
};

export default App;