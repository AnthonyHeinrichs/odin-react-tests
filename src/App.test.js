import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("App component", () => {
  it("renders Vancouver Canucks are going to make the playoffs", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders just kidding after button click", () => {
    render(<App />);
    const button = screen.getByRole("button", { name: "Click Me" });

    userEvent.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/Just kidding/i);
  });
});
