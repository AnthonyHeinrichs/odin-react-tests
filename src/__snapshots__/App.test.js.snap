// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders Vancouver Canucks are going to make the playoffs 1`] = `
<div>
  <button
    type="button"
  >
    Click Me
  </button>
  <h1>
    Vancouver Canucks are going to the playoffs
  </h1>
</div>
`;
